python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -497,3 +497,33 @@ def transfer_datasets(**kwargs):
     """
     number = script.transferDatasets(**kwargs)
     print(f"Number of datasets transferred: {number}")
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@click.argument("parent", required=True, nargs=1)
+@click.argument("children", required=False, nargs=-1)
+@click.option("--pop", is_flag=True, default=False,
+              help="Pop the first collection off the chain. Can not be used if CHILDREN are given.")
+@click.option("--doc", default="",
+              help="Documentation string associated with this collection. "
+              "Only relevant if the collection is newly created.")
+@click.option("--flatten/--no-flatten", default=False,
+              help="If `True` recursively flatten out any nested chained collections in children first.")
+@click.option("--mode",
+              type=click.Choice(["redefine", "extend", "remove"]),
+              default="redefine",
+              help="Update mode: "
+              "'redefine': Create new chain or redefine existing chain with the supplied CHILDREN. "
+              "'remove': Modify existing chain to remove the supplied CHILDREN. "
+              "'extend': Modify existing chain to extend it with the supplied CHILDREN.")
```

Comment:
Would `--mode=prepend` be useful?

Reply:
Yes, and more useful than `extend`, I think.  We may represent these things as Sequences in Python, but conceptually we (`pipetask` in particular) really treat them as LIFO stacks, where all of the mutations happen at the front (yeah, maybe I should have implemented them as reversed-lists in Python, but I figured ordering the container according to the search order is more important).

So I really did mean for `--pop` to act on the front, and I really only care that we provide ways to act on the front in general, but I could easily believe that also providing ways to act on the end is the best way to make it clear what all of the actions actually do.